PorePranav/CampusUnify
# Add Docker Configuration for Frontend and Backend Services

### Fixes #14 

### Description:

This PR adds Docker support for both the frontend (Vite) and backend (Node.js) services in the project. The Dockerfiles and docker-compose.yml are configured to streamline local development and deployment, enabling the services to run in isolated containers.

### Changes Introduced:

- Dockerfiles:
- Created a Dockerfile in the client folder for the frontend (Vite) application.
- Created a Dockerfile in the server folder for the backend (Node.js) application.
- Docker Compose Configuration:
- Added docker-compose.yml to manage both frontend and backend services.
- Configured services to run on:
- Frontend: Port 5173 for Vite development server.
- Backend: Dynamic port mapping using BACKEND_PORT from environment variables, with a default fallback to 3000.
- Both services are connected via a custom Docker network for internal communication.

### Notes:

- The backend is throwing an exception related to missing RAZORPAY_KEY_ID and RAZORPAY_KEY_SECRET. This error is unrelated to the Docker setup and should be addressed separately by configuring the necessary environment variables.

### How to Test:

1. Ensure Docker and Docker Compose are installed on your machine.
2. Run docker-compose up --build from the root directory.
3. Access the frontend via http://localhost:5173.
4. The backend should run on the port specified by BACKEND_PORT (default is 3000).
